<launch>


<machine name="master" address="localhost"/>

<!--   Kinect driver-->
  <arg name="fps_limit_kinect"    default="20"/>
  <arg name="max_depth"           default="9.0"/>
  <arg name="min_depth"           default="0.5"/>

<!-- Enable/disable stuff-->
  <arg name="run_peopledetector"  default="false"/>
  <arg name="run_compressor_remote"      default="false"/>
  <arg name="run_compressor_local"      default="true"/>


<!--  Compressor parameters-->
  <arg name="filter_value"        default="80" /> <!--decompressor [0-255]-->
  <arg name="voxel_size_small" default="0.12"/>
  <arg name="voxel_size_big" default="0.12"/>
  <arg name="voxel_size" default="0.04"/>
  <arg name="crop"      default="true"/>
  <arg name="min_x"     default="0.5"/>
  <arg name="max_x"     default="9.5"/>
  <arg name="min_y"     default="0.5"/>
  <arg name="max_y"     default="9.5"/>
  <!--<arg name="min_z"     default="0.1"/>-->
  <arg name="min_z"     default="-0.1"/>
  <arg name="max_z"     default=" 4.3"/>


<!-- ================= Compressor (local) ======================= -->
<group if="$(arg run_compressor_local)">

<node pkg="wp3_compressor" type="wp3_compressor_node" name="p26_jetson1_comp_kinect_1" machine="master">
    <param name="resolution" value="$(arg voxel_size)" />
    <param name="input_topic" value="/master/jetson1/points_nocolor" />
    <param name="input_type" value="0" />
    <param name="output_topic" value="/p26_lefty/jetson1/wp3/kinect_comp" />
    <param name="local_frame" value="jetson1_ir_optical_frame" />
    <param name="global_frame" value="world" />
    <param name="crop"      value="$(arg crop)" />
    <param name="min_x"     value="$(arg min_x)" />
    <param name="max_x"     value="$(arg max_x)" />
    <param name="min_y"     value="$(arg min_y)" />
    <param name="max_y"     value="$(arg max_y)" />
    <param name="min_z"     value="$(arg min_z)" />
    <param name="max_z"     value="$(arg max_z)" />
</node>

<node pkg="wp3_compressor" type="wp3_compressor_node" name="p26_jetson2_comp_kinect_1" machine="master">
    <param name="resolution" value="$(arg voxel_size)" />
    <param name="input_topic" value="/master/jetson2/points_nocolor" />
    <param name="input_type" value="0" />
    <param name="output_topic" value="/p26_lefty/jetson2/wp3/kinect_comp" />
    <param name="local_frame" value="jetson2_ir_optical_frame" />
    <param name="global_frame" value="world" />
    <param name="crop"      value="$(arg crop)" />
    <param name="min_x"     value="$(arg min_x)" />
    <param name="max_x"     value="$(arg max_x)" />
    <param name="min_y"     value="$(arg min_y)" />
    <param name="max_y"     value="$(arg max_y)" />
    <param name="min_z"     value="$(arg min_z)" />
    <param name="max_z"     value="$(arg max_z)" />
</node>

<node pkg="wp3_compressor" type="wp3_compressor_node" name="p26_jetson3_comp_kinect_1" machine="master">
    <param name="resolution" value="$(arg voxel_size)" />
    <param name="input_topic" value="/master/jetson3/points_nocolor" />
    <param name="input_type" value="0" />
    <param name="output_topic" value="/p26_lefty/jetson3/wp3/kinect_comp" />
    <param name="local_frame" value="jetson3_ir_optical_frame" />
    <param name="global_frame" value="world" />
    <param name="crop"      value="$(arg crop)" />
    <param name="min_x"     value="$(arg min_x)" />
    <param name="max_x"     value="$(arg max_x)" />
    <param name="min_y"     value="$(arg min_y)" />
    <param name="max_y"     value="$(arg max_y)" />
    <param name="min_z"     value="$(arg min_z)" />
    <param name="max_z"     value="$(arg max_z)" />
</node>

<node pkg="wp3_compressor" type="wp3_compressor_node" name="p26_jetson4_comp_kinect_1" machine="master">
    <param name="resolution" value="$(arg voxel_size)" />
    <param name="input_topic" value="/master/jetson4/points_nocolor" />
    <param name="input_type" value="0" />
    <param name="output_topic" value="/p26_lefty/jetson4/wp3/kinect_comp" />
    <param name="local_frame" value="jetson4_ir_optical_frame" />
    <param name="global_frame" value="world" />
    <param name="crop"      value="$(arg crop)" />
    <param name="min_x"     value="$(arg min_x)" />
    <param name="max_x"     value="$(arg max_x)" />
    <param name="min_y"     value="$(arg min_y)" />
    <param name="max_y"     value="$(arg max_y)" />
    <param name="min_z"     value="$(arg min_z)" />
    <param name="max_z"     value="$(arg max_z)" />
</node>

<node pkg="wp3_compressor" type="wp3_compressor_node" name="p26_jetson5_comp_kinect_1" machine="master">
    <param name="resolution" value="$(arg voxel_size)" />
    <param name="input_topic" value="/master/jetson5/points_nocolor" />
    <param name="input_type" value="0" />
    <param name="output_topic" value="/p26_lefty/jetson5/wp3/kinect_comp" />
    <param name="local_frame" value="jetson5_ir_optical_frame" />
    <param name="global_frame" value="world" />
    <param name="crop"      value="$(arg crop)" />
    <param name="min_x"     value="$(arg min_x)" />
    <param name="max_x"     value="$(arg max_x)" />
    <param name="min_y"     value="$(arg min_y)" />
    <param name="max_y"     value="$(arg max_y)" />
    <param name="min_z"     value="$(arg min_z)" />
    <param name="max_z"     value="$(arg max_z)" />
</node>

<node pkg="wp3_compressor" type="wp3_compressor_node" name="p26_jetson6_comp_kinect_1" machine="master">
    <param name="resolution" value="$(arg voxel_size)" />
    <param name="input_topic" value="/master/jetson6/points_nocolor" />
    <param name="input_type" value="0" />
    <param name="output_topic" value="/p26_lefty/jetson6/wp3/kinect_comp" />
    <param name="local_frame" value="jetson6_ir_optical_frame" />
    <param name="global_frame" value="world" />
    <param name="crop"      value="$(arg crop)" />
    <param name="min_x"     value="$(arg min_x)" />
    <param name="max_x"     value="$(arg max_x)" />
    <param name="min_y"     value="$(arg min_y)" />
    <param name="max_y"     value="$(arg max_y)" />
    <param name="min_z"     value="$(arg min_z)" />
    <param name="max_z"     value="$(arg max_z)" />
</node>

</group>

<!-- ================= Decompressor (local) ======================= -->
<!-- ================= Jetson1 ======================= -->
<node pkg="p26_decompressor" type="p26_decompressor_node" name="p26_jetson1_decomp_kinect">
<param name="input_topic" value="/p26_lefty/jetson1/wp3/kinect_comp" />
<param name="output_topic" value="/p26_lefty/jetson1/wp3/kinect_decomp" />
<param name="sensor_name" value="jetson1" />
<!--    <param name="sensor_frame" value="p26_jetson1_ir_optical_frame" />-->
    <param name="sensor_frame" value="world" />
    <param name="sensor_type" value="0" />
    <param name="filter_value" value="$(arg filter_value)" />
</node>
<!-- ================= Jetson2 ======================= -->
<node pkg="p26_decompressor" type="p26_decompressor_node" name="p26_jetson2_decomp_kinect">
<param name="input_topic" value="/p26_lefty/jetson2/wp3/kinect_comp" />
<param name="output_topic" value="/p26_lefty/jetson2/wp3/kinect_decomp" />
<param name="sensor_name" value="jetson2" />
<!--    <param name="sensor_frame" value="p26_jetson2_ir_optical_frame" />-->
    <param name="sensor_frame" value="world" />
    <param name="sensor_type" value="0" />
    <param name="filter_value" value="$(arg filter_value)" />
</node>
<!-- ================= Jetson3 ======================= -->
<node pkg="p26_decompressor" type="p26_decompressor_node" name="p26_jetson3_decomp_kinect">
<param name="input_topic" value="/p26_lefty/jetson3/wp3/kinect_comp" />
<param name="output_topic" value="/p26_lefty/jetson3/wp3/kinect_decomp" />
<param name="sensor_name" value="jetson3" />
<!--    <param name="sensor_frame" value="p26_jetson3_ir_optical_frame" />-->
    <param name="sensor_frame" value="world" />
    <param name="sensor_type" value="0" />
    <param name="filter_value" value="$(arg filter_value)" />
</node>
<!-- ================= Jetson4 ======================= -->
<node pkg="p26_decompressor" type="p26_decompressor_node" name="p26_jetson4_decomp_kinect">
<param name="input_topic" value="/p26_lefty/jetson4/wp3/kinect_comp" />
<param name="output_topic" value="/p26_lefty/jetson4/wp3/kinect_decomp" />
<param name="sensor_name" value="jetson4" />
<!--    <param name="sensor_frame" value="p26_jetson4_ir_optical_frame" />-->
    <param name="sensor_frame" value="world" />
    <param name="sensor_type" value="0" />
    <param name="filter_value" value="$(arg filter_value)" />
</node>
<!-- ================= Jetson5 ======================= -->
<node pkg="p26_decompressor" type="p26_decompressor_node" name="p26_jetson5_decomp_kinect">
<param name="input_topic" value="/p26_lefty/jetson5/wp3/kinect_comp" />
<param name="output_topic" value="/p26_lefty/jetson5/wp3/kinect_decomp" />
<param name="sensor_name" value="jetson5" />
    <param name="sensor_frame" value="world" />
<!--    <param name="sensor_frame" value="p26_jetson5_ir_optical_frame" />-->
    <param name="sensor_type" value="0" />
    <param name="filter_value" value="$(arg filter_value)" />
</node>
<!-- ================= Jetson6 KINECT ======================= -->
<node pkg="p26_decompressor" type="p26_decompressor_node" name="p26_jetson6_decomp_kinect">
<param name="input_topic" value="/p26_lefty/jetson6/wp3/kinect_comp" />
<param name="output_topic" value="/p26_lefty/jetson6/wp3/kinect_decomp" />
<param name="sensor_name" value="jetson6" />
<!--    <param name="sensor_frame" value="p26_jetson6_ir_optical_frame" />-->
    <param name="sensor_frame" value="world" />
    <param name="sensor_type" value="0" />
    <param name="filter_value" value="$(arg filter_value)" />
</node>

<!-- ================= J/transform_nodelet1etson7 KINECT ======================= -->
<!--<node pkg="p26_decompressor" type="p26_decompressor_node" name="jetson7_decomp_kinect">
    <param name="sensor_name" value="jetson7" />
    <param name="sensor_type" value="0" />
    <param name="filter_value" value="$(arg filter_value)" />
</node>-->
<!-- ================= Jetson6 VELODYNE ======================= -->
<!--<node pkg="p26_decompressor" type="p26_decompressor_node" name="jetson6_decomp_velodyne">
    <param name="sensor_name" value="jetson6" />
    <param name="sensor_type" value="1" />
    <param name="filter_value" value="$arg filter_value" />
</node>-->


<!-- ================= Compressor (local) ======================= -->
<group if="$(arg run_compressor_local)">

<node pkg="wp3_compressor" type="wp3_compressor_node" name="p26_jetson_comp_kinect" machine="master">
    <param name="resolution" value="$(arg voxel_size)" />
    <param name="input_topic" value="/p26_lefty/kinect_decomp_merged" />
    <param name="input_type" value="0" />
    <param name="output_topic" value="/p26_lefty/wp3/kinect_comp_merged_filtered" />
    <param name="local_frame" value="world" />
    <param name="global_frame" value="world" />
    <param name="crop"      value="$(arg crop)" />
    <param name="min_x"     value="$(arg min_x)" />
    <param name="max_x"     value="$(arg max_x)" />
    <param name="min_y"     value="$(arg min_y)" />
    <param name="max_y"     value="$(arg max_y)" />
    <param name="min_z"     value="$(arg min_z)" />
    <param name="max_z"     value="$(arg max_z)" />
</node>

</group>


<!-- ================= Decompressor (local) ======================= -->
<node pkg="p26_decompressor" type="p26_decompressor_node" name="p26_jetson_decomp_kinect">
    <param name="input_topic" value="/p26_lefty/wp3/kinect_comp_merged_filtered" />
    <param name="output_topic" value="/p26_lefty/point_cloud_obstacles" />
<!--    <param name="sensor_frame" value="p26_jetson1_ir_optical_frame" />-->
    <param name="sensor_frame" value="world" />
    <param name="sensor_type" value="0" />
    <param name="filter_value" value="$(arg filter_value)" />
</node>

</launch>
